session.go

```diff
@@ -457,22 +456,37 @@ func (s *sessionImpl) StartTunnel(ctx context.Context, cfg TunnelConfig) (Tunnel
 	}, nil
 }
 
-type SrvInfo proto.SrvInfoResp
-type AuthResp proto.AuthResp
+// The rest of the `sessionImpl` methods are non-public, but can be
+// interface-asserted if they're *really* needed. These are exempt from any
+// stability guarantees and subject to change without notice.
 
-func (s *sessionImpl) AuthResp() AuthResp {
-	return AuthResp(s.inner().AuthResp)
+func (s *sessionImpl) ProtoVersion() string {
+	return s.inner().ProtoVersion
 }
-
-func (s *sessionImpl) SrvInfo() (SrvInfo, error) {
-	resp, err := s.inner().SrvInfo()
-	return SrvInfo(resp), err
+func (s *sessionImpl) ServerVersion() string {
+	return s.inner().ServerVersion
+}
+func (s *sessionImpl) ClientID() string {
+	return s.inner().ClientID
+}
+func (s *sessionImpl) AccountName() string {
+	return s.inner().AccountName
+}
+func (s *sessionImpl) PlanName() string {
+	return s.inner().PlanName
+}
+func (s *sessionImpl) Banner() string {
+	return s.inner().Banner
+}
+func (s *sessionImpl) SessionDuration() int64 {
+	return s.inner().SessionDuration
+}
+func (s *sessionImpl) Region() string {
```

Comment:
region feels potentially interesting for anyone who uses the geo-balanced dns name.

Is that info available with a better ux somewhere?

Reply:
Yeah, I agree. I'm just not sure where it would go.

I can think of a few approaches, none of which are great:
* Add some of these getters to the `Session` interface
* Add an `Info` method to the `Session` interface that returns a collection of these properties
* Return a concrete type rather than the interface, and expose them there

The first two dirty up the interface and are counter to the "Just `Close` and `StartTunnel`" goal, and the later makes it harder for us to "hide" things from the public API, but still keep them kinda accessible.

Reply:
`Session.Metadata()` doesn't seem _that bad_.

But also, we could add it as an `OnConnect` interceptor, i.e. have that pass in `Session, SessionMetadata` as arguments. It's fixed for a session, so it can just be yanked out there reasonably.

I'm happy merging this as-is and iterating further if the need arises.

Reply:
An interceptor argument seems interesting!

I'll make a note of these ideas for discussion next time we get everyone in a room for a review.